furi/core/string.c

```diff
@@ -258,7 +259,37 @@ void furi_string_set_n(FuriString* v, const FuriString* ref, size_t offset, size
 }
 
 size_t furi_string_utf8_length(FuriString* str) {
-    return string_length_u(str->string);
+    const char * cstr = string_get_cstr(str->string);
+
+    size_t len = 0;
+    int8x4_t zero = 0x00000000;
+    int8x4_t one  = 0x01010101;
+    int8x4_t threshold = -1077952577; // -65, -65, -65, -65
+
+    for (;;) {
+        uint8x4_t result;
+        int8x4_t vec = *(int8x4_t *) cstr;
+
+        __uadd8(0xFFFFFFFF, vec); result = __sel(zero, one);
+        if (result > 0) {
+            if (result == 0x01000000) {
```

Comment:
This piece of code can be removed. It doesn't give a good performance boost, but the code gets more complicated.